File: app/src/main/java/com/tistory/black_jin0427/myandroidarchitecture/view/dratil/DetailActivity.java
package com.tistory.black_jin0427.myandroidarchitecture.view.dratil;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.tistory.black_jin0427.myandroidarchitecture.BaseActivity;
import com.tistory.black_jin0427.myandroidarchitecture.R;
import com.tistory.black_jin0427.myandroidarchitecture.api.model.User;
import com.tistory.black_jin0427.myandroidarchitecture.rxEventBus.RxEvent;

import org.w3c.dom.Text;

import butterknife.BindView;
import butterknife.OnClick;
import de.hdodenhof.circleimageview.CircleImageView;

public class DetailActivity extends BaseActivity {

    @BindView(R.id.iv_detail_profile)
    CircleImageView ivDetailProfile;

    @BindView(R.id.tv_detail_like_cnt)
    TextView tvDetailLIkeCnt;

    public final static String KEY_USER = "key_user";

    private User user;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);
        getUserFromIntent();
    }

    private void getUserFromIntent(){
        user = (User) getIntent().getSerializableExtra(KEY_USER);
        setTitle(user.getFullName());
        initData(user);
    }

    private void initData(User user) {

        tvDetailLIkeCnt.setText(user.getLikeCnt());

        Glide.with(this)
                .load(user.picture.large)
                .into(ivDetailProfile);

    }

    @OnClick(R.id.btn_detail_like) void onClick() {
        user.likeCnt++;
        tvDetailLIkeCnt.setText(user.getLikeCnt());

        RxEvent.getInstance().sendEvent(user);
    }
}

File: app/src/main/java/com/tistory/black_jin0427/myandroidarchitecture/view/DetailActivity.kt
package com.tistory.black_jin0427.myandroidarchitecture.view

import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import com.tistory.black_jin0427.myandroidarchitecture.BaseActivity
import com.tistory.black_jin0427.myandroidarchitecture.R
import com.tistory.black_jin0427.myandroidarchitecture.api.model.User
import com.tistory.black_jin0427.myandroidarchitecture.databinding.ActivityDetailBinding
import com.tistory.black_jin0427.myandroidarchitecture.viewModel.UserViewModel

class DetailActivity: BaseActivity<ActivityDetailBinding>() {

    companion object {

        val KEY_USER = "key_user"
    }

    override val layoutResourceId = R.layout.activity_detail

    private lateinit var userViewModel: UserViewModel

    private lateinit var user: User

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // 뷰 모델 초기화 및 관찰자 연결
        userViewModel = ViewModelProviders.of(this).get(UserViewModel::class.java)

        // 데이터 바인딩에 뷰 모델 연결
        viewDataBinding.user = userViewModel

        // 데이터 바인딩에 LifecycleOwner 연결하여 liveData 를 DataBinding 과 사용 할 수 있게 함
        viewDataBinding.setLifecycleOwner(this)

        // 데이터 초기화
        getUserFromIntent()
    }

    private fun getUserFromIntent() {
        user    = intent.getSerializableExtra(KEY_USER) as User
        title   = user.fullName

        setData()
    }

    private fun setData() {

        userViewModel.setUser(user)

    }
}
